src/App.tsx

```diff
@@ -8,48 +8,62 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+import { getUsers } from './api/users';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as usersActions } from './features/usersSlice';
+import { actions as authorActions } from './features/authorSlice';
+import { actions as postsActions } from './features/postsSlice';
+import { actions as selectedPostActions } from './features/selectedPostSlice';
 import { User } from './types/User';
 import { Post } from './types/Post';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const author = useAppSelector(state => state.author);
+  const { posts, loading, hasError } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.selectedPost);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => dispatch(usersActions.set(usersFromServer)));
+  }, []);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectedPostActions.remove());
 
     if (author) {
-      loadUserPosts(author.id);
+      dispatch(postsActions.init(author.id));
     } else {
-      setPosts([]);
+      dispatch(postsActions.clear());
     }
   }, [author]);
 
+  const showLoader = author && loading;
+  const showError = author && !loading && hasError;
+  const showNoPostsYet = author && !loading && !hasError && posts.length === 0;
+  const showPostList = author && !loading && !hasError && posts.length > 0;
+
+  const setAutor = useCallback(
+    (currAuthor: User) => dispatch(authorActions.set(currAuthor)),
+    [],
```

Comment:
please do not ignore this eslint warning as it is very important

Reply:
In the redux documentation, it is written that [the dispatch function will not change if we do not change the store](https://react-redux.js.org/api/hooks#:~:text=The%20dispatch%20function%20reference,to%20do%20just%20that%3A). And one of the three main principles of redux is 'Single source of truth'. From the documentation: ['The global state of your application is stored in an object tree within a single store.'](https://redux.js.org/understanding/thinking-in-redux/three-principles).

So we just pass unnecessary dependencies to useEffect() and waste more resources, they are not important. By the same example, the function that we should insert, but we don't insert, because the linter understands that it is not needed is setSomeState().